### ISSUE TITLE
Action Callback not being called

### ISSUE THREAD

**Issue by Maximvdw** [phonegap/phonegap-plugin-push#466]
Dear,

I am using the latest version:
phonegap-plugin-push 1.5.2 "PushPlugin"

In my application (main index.js)

    window.allowEntrance = function (data) { 
        console.log('Allowed');
    	alert(data);
    };
    window.denyEntrance = function (data) { 
        console.log('Denied');
    	alert(data);
    };

My GCM notification has two action buttons with their callbacks being
"window.allowEntrance"
and
"window.denyEntrance"

In the ADB Monitor I see this:
`12-30 20:50:44.855 26309-26393/? D/PushPlugin_GCMIntentService: adding callback = window.allowEntrance`
`12-30 20:50:44.859 26309-26393/? D/PushPlugin_GCMIntentService: adding callback = window.denyEntrance`

however, when I click the buttons the alert is not shown (or nothing in the console) and the notification remains.

Regards,
Maxim

**Comment by macdonst** [comment]
@Maximvdw try using "allowEntrance" and "denyEntrance" as your callbacks. That is when you send the payload via GCM omit the "window." prefix.

**Comment by Maximvdw** [comment]
Thanks for the quick response.

My code is now as follows:

    function allowEntrance(data) { 
    	console.log('Allowed');
        alert(data);
    }
    function denyEntrance(data) { 
        console.log('Denied');
    	alert(data);
    }


My ADB shows:
`12-31 13:19:18.347 12476-12495/? D/PushPlugin_GCMIntentService: adding callback = denyEntrance`
`12-31 13:19:18.342 12476-12498/? D/PushPlugin_GCMIntentService: adding callback = allowEntrance`

I see no errors when clicking the buttons, however when clicking them -> the application opens -> nothing happens (the notification also remains on screen)

- Maxim

**Comment by Maximvdw** [comment]
I solved the error, my mistake. While looking through the whole ADB log I saw that the plugin cached the extras but I did not see it being 'executed'.

So apparently my fault was that I only initialized the pushplugin after a login screen (to also register the device using the user credentials). But that meant that the plugin wouldn't execute those callbacks after a log in

This issue may be marked solved, sorry for wasting your time ;)

Regards,
Maximvdw
